openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceDetailsPage.tsx

```diff
@@ -962,21 +1016,59 @@ const ServiceDetailsPage: FunctionComponent = () => {
     typeFilter,
   ]);
 
+  useEffect(() => {
+    if (isCollateAIWidgetSupported) {
+      fetchCollateAgentsList({
+        limit: collateAgentPagingCursor?.pageSize ?? collateAgentPageSize,
+      });
+    }
+  }, [collateAgentPageSize]);
```

Comment:
```suggestion
  }, [collateAgentPageSize, isCollateAIWidgetSupported]);
```

Reply:
The useEffect dependency does not need `isCollateAIWidgetSupported`